src/main/java/com/teragrep/cfe_39/consumers/kafka/HDFSPrune.java

```diff
@@ -0,0 +1,139 @@
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+   http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+ */
+
+package com.teragrep.cfe_39.consumers.kafka;
+
+import com.teragrep.cfe_39.Config;
+import org.apache.hadoop.conf.Configuration;
+import org.apache.hadoop.fs.FileStatus;
+import org.apache.hadoop.fs.FileSystem;
+import org.apache.hadoop.fs.LocalFileSystem;
+import org.apache.hadoop.fs.Path;
+import org.apache.hadoop.hdfs.DistributedFileSystem;
+import org.apache.hadoop.security.UserGroupInformation;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+import java.net.URI;
+import java.util.Properties;
+import java.util.concurrent.TimeUnit;
+
+public class HDFSPrune {
+    private static final Logger LOGGER = LoggerFactory.getLogger(HDFSPrune.class);
+    private Config config;
+    private final FileSystem fs;
+    private Path newFolderPath;
+    private long cutoff_epoch;
+    private final boolean useMockKafkaConsumer; // test-mode switch
+
+    public HDFSPrune(Config config, String topicName) throws IOException {
+
+        // Check for testmode from config.
+        Properties readerKafkaProperties = config.getKafkaConsumerProperties();
+        this.useMockKafkaConsumer = Boolean.parseBoolean(
+                readerKafkaProperties.getProperty("useMockKafkaConsumer", "false")
+        );
+
+        if (useMockKafkaConsumer) {
+            this.config = config;
+            String hdfsuri = config.getHdfsuri();
+
+            String path = config.getHdfsPath() + "/" + topicName;
+            // ====== Init HDFS File System Object
+            Configuration conf = new Configuration();
+            // Set FileSystem URI
+            conf.set("fs.defaultFS", hdfsuri);
+            // Because of Maven
+            conf.set("fs.hdfs.impl", org.apache.hadoop.hdfs.DistributedFileSystem.class.getName());
+            conf.set("fs.file.impl", org.apache.hadoop.fs.LocalFileSystem.class.getName());
+            // Set HADOOP user
+            System.setProperty("HADOOP_USER_NAME", "hdfs");
+            System.setProperty("hadoop.home.dir", "/");
+            //Get the filesystem - HDFS
+            fs = FileSystem.get(URI.create(hdfsuri), conf);
+
+            //==== Create folder if not exists
+            Path workingDir = fs.getWorkingDirectory();
+            newFolderPath = new Path(path);
+            if (!fs.exists(newFolderPath)) {
+                // Create new Directory
+                fs.mkdirs(newFolderPath);
+                LOGGER.info("Path " + path + " created.");
```

Comment:
avoid string concat in logger: use `LOGGER.info("Path {} created.", path)` instead.